.github/workflows/ci.yml

```diff
@@ -1,6 +1,13 @@
 name: Keycloak CI
 
-on: [push, pull_request]
+on:
```

Comment:
@andreaTP if I understood correctly what is described on #10913 by @stianst, we will run the workflows on nightly basis on push events, not on pull events.

I'd suggest something like:

```
on:
  # Run the workflows on nightly basis on push
  push:
  schedule:
    - cron: '0 0 * * *'
    
  pull_request:
```

Reply:
@abstractj if I understand your comment correctly there is no behavioural difference from the two signatures in GH Action, as they are simply separate elements in an array:
```yaml
on:
  push:
  schedule:
  pull:
```
vs:
```yaml
on:
  pull:
  schedule:
  push:
```

e.g. each element of the array is independent from the others.

reference docs: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#using-multiple-events

Reply:
@andreaTP you are correct, here is my updated suggestion:

```
on:
  schedule:
    - cron: '0 0 * * *'
    
  pull_request:
```
Ignoring only the `main` branch, does not prevent that changes in the branch like `17.0` won't be triggered. Completely removing it will do the work. If it's our intention to trigger the builds on every change on backport branches, please ignore my comments.

For pull-request, we want to trigger pull-request not only targeting the `main` branch, but also when pull-request are submitted to branches created exclusively to backport issues, like 17.0.

We have an issue for this https://github.com/keycloak/keycloak/issues/10845. Does it make sense?

Reply:
I briefly touched on this point with @stianst , we discussed on the fact that the current implementation is more friendly to development workflow in forks, i.e.:

Removing `push` at all:
 - to run a CI in your own fork you always need to change the CI job (revert the change back before submitting a PR and squash commits)
 
 Keeping the `push` but ignoring `main`:
 - comply with the requirement in #10913
 - enable to run CIs in fork repositories without the need for extra tweaking of the CI

As most things in software this is a trade-off, I'm pretty much convinced that keeping a good developer experience outweighs the extra jobs running in the rare occasions of backports.

Reply:
+1